### PR TITLE
Fixes #26 Textarea now respects rows attribute

### LINKED ISSUE #26: .textarea doesn't honor <textarea>'s `rows` attribute
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-rows

### PR DESCRIPTION
**Pull request by timacdonald**
### Proposed solution
This pull request fixes issue #26. Previously the `.textarea` class would override the ability to set the height to the `<textarea>` tag using the `rows="2"` attribute. I've added a CSS selector to detect if the `.textarea` has the `rows` attribute and in it - reset all height related properties to their default values.

Properties include: `height: auto`, `min-height: 0`, and `max-height: none`

### Tradeoffs
- Unfortunately this has to override the values as I am unaware of anyway to create a CSS selector to target **.textarea classes that do not have the `rows` attribute**. If this is possible, we would not have to override - but alas.
- This is a breaking change for those with rows attribute defined.
- It is not a **purely** class based solution - so you may not be a fan.

### Testing Done
[Working CodePen](https://codepen.io/timacdonald/pen/zzyRGx)

If you are keen to have this in Bulma, do you think we should do the same for the `cols` attribute.

### CODE REVIEW

**Comment by jdanyow** [comment]
@timacdonald would `.textarea:not([rows])` work re: *create a CSS selector to target .textarea classes that do not have the rows attribute*.

**Comment by timacdonald** [comment]
huh...yea...it totally does! That never crossed my mind. I'm going to tweak this based on that. thanks @jdanyow!

**Comment by timacdonald** [comment]
So, I'm not sure if this is now a better solution. It is a little bit more convoluted than the original solution. Looking at the applied CSS rules I think this is a better solution, but looking at the SASS this new approach is a little bit more complicated.

## Original

Just added this rule, but it meant that we were overriding rules, which I think is best if you can keep that to a minimum.

```sass
.textarea[rows]
  height: auto;
  min-height: 0;
  max-height: none;
```

## The current PR

This approach feels better from an outputted CSS perspective as there is less overriding of rules.

Thoughts?

File: sass/elements/form.sass [hunk 1]
```diff
@@ -70,13 +70,18 @@ $input-radius:              $radius !default
...
 .textarea
   display: block
-  max-height: 600px
   max-width: 100%
-  min-height: 120px
   min-width: 100%
   padding: 0.625em
   resize: vertical
 
+.textarea:not([rows])
```
Comment (jgthms) [inline sass/elements/form.sass]: This selector should be `&:not([rows])`

File: sass/elements/form.sass [hunk 2]
```diff
@@ -70,13 +70,18 @@ $input-radius:              $radius !default
...
+  max-height: 600px
+  min-height: 120px
+
+.textarea[rows]
+  height: auto
```
Comment (jgthms) [inline sass/elements/form.sass]:
```
&[rows]
  height: unset
```

**Comment by timacdonald** [comment]
Yea...sorry about that - new to SASS. I've updated those selectors for ya.

**Comment by jgthms** [comment]
Thanks!

**Comment by timacdonald** [comment]
No, thank you @jgthms 👍

**Comment by omgaz** [comment]
Thanks @timacdonald :) Have a cookie of appreciation 🍪

**Comment by timacdonald** [comment]
I shall treasure this 🍪 forever and never wash it! Thanks @omgaz for the cookie love.

**Pull request merged**